tests/rsa_tests.rs

```diff
@@ -142,24 +142,19 @@ fn test_signature_rsa_pkcs1_sign_output_buffer_len() {
         include_bytes!("../src/rsa/signature_rsa_example_private_key.der");
     let key_pair = rsa::KeyPair::from_der(PRIVATE_KEY_DER).unwrap();
 
-    // The output buffer is one byte too short.
-    let mut signature = vec![0; key_pair.public().modulus_len() - 1];
-
-    assert!(key_pair
-        .sign(&signature::RSA_PKCS1_SHA256, &rng, MESSAGE, &mut signature)
-        .is_err());
-
-    // The output buffer is the right length.
-    signature.push(0);
-    assert!(key_pair
-        .sign(&signature::RSA_PKCS1_SHA256, &rng, MESSAGE, &mut signature)
-        .is_ok());
-
-    // The output buffer is one byte too long.
-    signature.push(0);
-    assert!(key_pair
-        .sign(&signature::RSA_PKCS1_SHA256, &rng, MESSAGE, &mut signature)
-        .is_err());
+    // Wen the output buffer is not exactly the right length, `sign()` returns
+    // an error (and does not panic or invoke UB). if `sign` doesn't check that
```

Comment:
Based on this comment, shouldn't there be some tests checking the error conditions like before?

Reply:
Sorry, I don't understand what you mean. Note that the test is testing that the condition "the buffer is the right length" is equal to (`assert_eq!`) the condition "the result is Ok(_).".

Reply:
Oh, this is my bad. I re-read the test and noticed that the left-hand side of the `assert_eq!()` is dependent on `len`'s value. I missed that on my first read-through and combined with the comment, was tripped up.